doc/sdk-cli-reference.md

```diff
@@ -134,6 +134,21 @@ pkg/apis/app/v1alpha1/
 
 ### openapi
```

Comment:
Side note: should we remove this doc entirely now that we auto-generate CLI docs under `doc/cli`?

Reply:
Yeah, probably.

@theishshah is there anything else we need to do before we remove it?

Reply:
If we are just deprecated the command then: 
IHMO:
- We should not remove/change the docs
- We just need to add on top of the func used by it which will be removed the comments that it is deprecated for have the func strikethrough as:

<img width="306" alt="Screenshot 2019-12-04 at 21 39 44" src="https://user-images.githubusercontent.com/7708031/70184218-4fc43c00-16df-11ea-8a0a-6f4af62b4ea2.png">

Comment: `// Deprecated: xpto`

- Also, we need to provide a doc with the info for users knowhow them can customize the projects to still be able to gen and use the open API.

Then, should not be better have a PR to deprecated and one to add the new command?

Reply:
We want existing users who have written scripts around `generate openapi` to still be able to use the command while they migrate to `generate crds`, but new users to use `generate crds`. By removing docs we can ensure visibility of `generate crds` without confusing anyone.

The "documentation" explaining deprecation and new behavior is [here](https://github.com/operator-framework/operator-sdk/pull/2276/files#diff-d4bad6c476b1ac95862b5e0a0b1a9195R55-R66).

I'm ok with a deprecation comment, like it is used [here](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.2.0/pkg/runtime/log/log.go#L20).

Reply:
I can put the GoDoc deprecation comments on the functions used for `generate openapi` but they won't serve much purpose since none of the functions are public. IMO the big reason to add the deprecation comments is to signal to users that the API is changing, but we're the only users of these private functions.

 This PR isn't really a Go code API deprecation. It's a deprecation of a CLI subcommand.

Reply:
Hi @joelanford,
 
The comments would be helpful to us know what should not so longer be used since will be removed. Regards the command, a notice output as it was done [here](https://github.com/kubernetes-sigs/kubebuilder/pull/1137#discussion_r343405975) for example would be more appropriate. 

c/c @estroz

Reply:
This PR includes a deprecation notice in the output when `generate openapi` is run. See [here](https://github.com/operator-framework/operator-sdk/pull/2276/files#diff-d4bad6c476b1ac95862b5e0a0b1a9195R55-R66).